src/comp.py

```diff
@@ -23,49 +25,57 @@ def __repr__(self):
 # whose name starts with 'D':
 
 print("Starts with D:")
-r = []  # TODO
+r = [human.name for human in humans if human.name[0] == "D"]  # TODO
+# r = [h.name for h in humans if h.name.startswith("D")]
 print(r)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 
 print("Ends with e:")
-r = []  # TODO
+r = [human.name for human in humans if human.name[-1] == "e"]  # TODO
+# r = [h.name for h in humans if h.name.endswith("e")]
 print(r)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 
 print("Starts between C and G, inclusive:")
-r = []  # TODO
+r = [h.name for h in humans if (h.name[0] >= "C" and h.name[0] <= "G")]  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-r = []  # TODO
+r = [human.age + 10 for human in humans]  # TODO
+# r = [(h.age + 10) for h in humans]
 print(r)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-r = []  # TODO
+r = [print(f"{human.name}-{human.age}") for human in humans]  # TODO
+# r = [f"{h.name}-{h.age}" forh in humans]
 print(r)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-r = []  # TODO
+r = [print(f"({human.name}, {human.age})") for human in humans if human.age >= 27 and human.age <= 32]  # TODO
+# r = [(h.name, h.age) for h in humans if (h.age >= 27 and h.age <= 32)]
 print(r)
```

Comment:
The print(r) here and in line 58 are causing your output to include "None" for each entry. I think this is because r already includes a print command.